mapie/time_series_regression.py

```diff
@@ -0,0 +1,325 @@
+        """
+        ``predict`` method correspond to the ``MapieRegressor``'s one with the
+        argument ``JAB_Like`` in more. In case  ``JAB_Like`` is ``False``,
+        predictions correspond to [6]. In case it is True, is is another
+        implementation inspired of [2] with wider PIs. The second method is
+        much slower because of PIwise optimization.
+
+        Parameters
+        ----------
+        JAB_Like : boolean, default False
+            Whether to use the implementation of [6] or an implementation
+            closer to [2].
+        """
+
+        # Checks
+        check_is_fitted(self, self.fit_attributes)
+        self._check_ensemble(ensemble)
+        alpha = cast(Optional[NDArray], check_alpha(alpha))
+        X = check_array(X, force_all_finite=False, dtype=["float64", "object"])
+        y_pred = self.single_estimator_.predict(X)
+        n = len(self.conformity_scores_)
+
+        if alpha is None:
+            return np.array(y_pred)
+        else:
+            alpha_np = cast(NDArray, alpha)
+            check_alpha_and_n_samples(alpha_np, n)
+
+            if (
+                (not JAB_Like)
+                or (self.method in ["naive", "base"])
+                or (self.cv == "prefit")
+            ):
+                # This version of predict corresponds to [6].
+                # Its PIs are closed to the oracle's ones.
+                betas_0 = np.full_like(alpha_np, np.nan, dtype=float)
+                for ind, _alpha in enumerate(alpha_np):
+                    betas = np.linspace(
+                        _alpha / (n + 1), _alpha, num=n + 1, endpoint=False
+                    )
+
+                    one_alpha_beta = masked_quantile(
+                        ma.masked_invalid(self.conformity_scores_),
+                        1 - _alpha + betas,
+                        axis=0,
+                        method="higher",
+                    )  # type: ignore
+
+                    beta = masked_quantile(
+                        ma.masked_invalid(self.conformity_scores_),
+                        betas,
+                        axis=0,
+                        method="lower",
+                    )  # type: ignore
+
+                    betas_0[ind] = betas[
+                        np.argmin(one_alpha_beta - beta, axis=1)
+                    ]
+
+                lower_quantiles = masked_quantile(
+                    ma.masked_invalid(self.conformity_scores_),
+                    betas_0,
+                    axis=0,
+                    method="lower",
+                ).T  # type: ignore
+                higher_quantiles = masked_quantile(
+                    ma.masked_invalid(self.conformity_scores_),
+                    1 - alpha_np + betas_0,
+                    axis=0,
+                    method="higher",
+                ).T  # type: ignore
+
+                if (self.method in ["naive", "base"]) or (self.cv == "prefit"):
+                    y_pred_low = np.column_stack(
```

Comment:
`y_pred_low = y_pred[:, np.newaxis] + lower_quantiles` does not work ?

Reply:
Done